src/solver/impls/petsc/petsc.cxx

```diff
@@ -708,7 +708,7 @@ PetscErrorCode solver_ijacobian(TS ts, BoutReal t, Vec globalin, Vec UNUSED(glob
   ierr = solver_rhsjacobian(ts, t, globalin, J, Jpre, f_data); CHKERRQ(ierr);
 
   ////// Save data for preconditioner
-  PetscSolver *solver = (PetscSolver*) f_data;
+  auto* solver = (PetscSolver*)f_data;
```

Comment:
Is there a reason this isn't `auto* solver = static_cast<PetscSolver*>(user_data);`, like the other solvers?

Reply:
Meant to circle back and update this, thanks!